app/Models/Contest.php

```diff
@@ -189,33 +182,45 @@ public function vote(User $user, ContestEntry $entry)
 
     public function entriesByType($user = null)
     {
-        if ($this->isBestOf()) {
-            if ($user === null) {
-                return [];
-            }
+        $entries = $this->entries()->with('contest');
 
-            $playmode = Beatmap::MODES[$this->extra_options['best_of']['mode'] ?? 'osu'];
-
-            // This will only returns beatmap entries a user has played
-            $entries = $this->entries()
-                ->whereIn('entry_url', function ($query) use ($playmode, $user) {
-                    $query->select('beatmapset_id')
-                        ->from('osu_beatmaps')
-                        ->where('osu_beatmaps.playmode', '=', $playmode)
-                        ->whereIn('beatmap_id', function ($query) use ($user) {
-                            $query->select('beatmap_id')
-                                ->from('osu_user_beatmap_playcount')
-                                ->where('user_id', '=', $user->user_id);
-                        });
-                });
-        } else {
-            $entries = $this->entries();
-            if ($this->show_votes) {
+        if ($this->show_votes) {
+            return Cache::remember("contest_entries_with_votes_{$this->id}", 5, function () use ($entries) {
                 $entries = $entries->with('user');
+
+                if ($this->isBestOf()) {
+                    $entries = $entries
+                        ->selectRaw('*')
+                        ->selectRaw('(SELECT FLOOR(SUM(`weight`)) FROM `contest_votes` WHERE `contest_entries`.`id` = `contest_votes`.`contest_entry_id`) AS votes_count')
+                        ->limit(50); // best of contests tend to have a _lot_ of entries...
```

Comment:
also re: limit; the best_of contests will have the full results released separately, whereas not so much for the other contests